Technolab.OnlineLibrary.Web/Controllers/AuthController.cs

```diff
@@ -60,13 +62,50 @@ private async Task PerformLogin(User user)
                 new Claim(type: Consts.Claim.FirstName, value: user.FirstName),
                 new Claim(type: Consts.Claim.LastName, value: user.LastName),
             };
-            var identity = new ClaimsIdentity(claims, CookieAuthenticationDefaults.AuthenticationScheme);            
-            
+            var identity = new ClaimsIdentity(claims, CookieAuthenticationDefaults.AuthenticationScheme);
+
             await HttpContext.SignInAsync(
                 CookieAuthenticationDefaults.AuthenticationScheme,
                 new ClaimsPrincipal(identity));
         }
 
         private ILibraryDbContextFactory ContextFactory { get; }
+
+        private bool VerifyPassword(string enteredPassword, string correctHashWithParams)
+        {
+            string[] split = correctHashWithParams.Split(":");
+
+            string correctHash = split[0];
+            byte[] salt = Convert.FromBase64String(split[1]);
+            int iterations = Convert.ToInt32(split[2]);
+            HashAlgorithmName hashAlgorithm = new HashAlgorithmName(split[3]);
+            int keySize = Convert.ToInt32(split[4]);
+
+            var enteredPasswordHash = Rfc2898DeriveBytes.Pbkdf2(enteredPassword, salt, iterations, hashAlgorithm, keySize);
+            return CryptographicOperations.FixedTimeEquals(enteredPasswordHash, Convert.FromHexString(correctHash));
+        }
+
+        private string GenerateHash(string password)
+        {
+            int keySize = 32;

```

Comment:
Generally speaking avoid magic string/numbers. If anything, these settings should come from the config file.

Reply:
You mean the Consts.cs file, right? I'll put the delimiter (":") there as well in that case, if there are no objections.